projects/knora-ui/src/lib/viewer/pipes/knoradate.pipe.ts

```diff
@@ -7,25 +7,33 @@ import { KnoraDate } from '@knora/api';
 export class KnoraDatePipe implements PipeTransform {
 
     // TODO: use the optional format parameter and return a string with that format
-    transform(value: KnoraDate, format?: string): string {
-        if (!(value instanceof KnoraDate)) {
+    transform(date: KnoraDate, format?: string): string {
+        if (!(date instanceof KnoraDate)) {
             console.error('Non-KnoraDate provided. Expected a valid KnoraDate');
             return '';
         }
 
-        if (value == null) {
+        if (!date) {
             console.error('Value is null. Expected a valid KnoraDate');
             return '';
         }
 
-        const formattedDate: Date = new Date();
-
-        // js date month is 0-based, so subtract 1
-        formattedDate.setFullYear(value.year, (value.month - 1), value.day);
-
-        const dateStr = formattedDate.toLocaleDateString();
+        // make this fancier
+        switch (format) {
```

Comment:
Could we support https://angular.io/api/common/DatePipe#pre-defined-format-options and a locale?

Reply:
I did a similar thing in https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/blob/develop/projects/jdnconvertible-calendar-date-adapter/src/lib/jdnconvertible-calendar-date-formats.ts for the DatePicker. It would be great if we could make this consistent.

Reply:
this is another implementation provided by Angular Material for using the DatePicker with MOmentJS: https://github.com/angular/components/blob/master/src/material-moment-adapter/adapter/moment-date-formats.ts

Reply:
to be figured out in https://dasch.myjetbrains.com/youtrack/issue/DSP-203